rfcs/0006-runtime-references-interface.md

```diff
@@ -44,6 +45,11 @@ its builtin hash scanning and just use the file. Both files are
 newline-separated lists of store paths. Parse errors in the
 `runtime-references` file, or non-existent store paths, fail the build.
 
+To avoid duplicating the default reference scanning behavior in Nix,
+we could also make a tiny tool part of Nix itself that wraps the
+`scanForReferences` API, and then expose that in the `<nix>` namespace
+so tooling can get the full path to that tool.
```

Comment:
I guess we'd need to give our builds access to the full set of string closures, etc. for this to work. Another mechanism could be something like `exportReferencesGraph` but available as a magic callback in an env var. So you'd set

`magicReferenceScanner = true;` in your `derivationStrict` call. That would give your derivation an env var that when invoked scans the path passed in for all the references that `scanForReferences` API looks for. It'd basically be a "partial application" of that function, except crossing one or two language boundaries.